VERSIONING.md

```diff
@@ -1,10 +1,57 @@
 Versioning and Branching in Konveyor Projects
 =============================================
 
-This applies to the following Konveyor Projects:
+This document outlines how we handle versioning of our projects in the Konveyor
+organization (Konveyor projects).
 
-* konveyor/tackle2-hub
-* ...
+Individual projects may have more specific rules or guidelines.
+
+# Overview
+
+Konveyor projects follow [Semantic Versioning](https://semver.org/). This means
+that the version number of a project is expressed in the format X.Y.Z, where:
+
+* X is the major version, representing significant changes and new features
+* Y is the minor version, representing new features and improvements
+* Z is the patch version, representing bug fixes and minor improvements
+
+*NOTE*: If the major release is 0, any minor release may contain breaking changes.
+
+For library projects (ie. tackle2-addon), these guarantees extend to all types
+and functions exposed in public APIs. These guarantees do *not* extend to types
+and functions not exposed in public APIs.
+
+For projects providing an API interface (ie. [tackle2-hub](https://github.com/konveyor/tackle2-hub)),
+these guarantees extend to all user accessible API endpoints.
+
+# Branching
+
+In order to maintain our guarantees two types of branches will be maintained:
+`main` and `release-X.Y`.
+
+* `main` branch is where all of the development happens. It is not guaranteed 
+  stable and may contain breaking changes.
+* `release-X.Y` where X.Y is the minor version for a release is considered
+  stable. These branches will receive backport fixes for critical issues, but no
+  new features will be added.
+
+# Releases
+
+All releases from the main branch will be marked as pre-release and be of the
+form `vX.Y.0-alpha.n`, where `X.Y` represents the next minor release and `n`
+follows Semantic Versioning precedence rules for pre-release. For example,
+if the next minor release is going to be 2.0, a pre-release from the main
+branch might be `v2.0.0-alpha.1` or `v2.0.0-alpha.2`, depending on the number of
+pre-releases that have been made so far. By marking releases from the main
+branch as pre-release, we aim to clearly indicate that the code may contain
+breaking changes and is not yet considered stable.
+
+
+Releases from `release-X.Y` branches contain stable code and always start with
```

Comment:
by marking a release, do we mean tagging a  particular commit on those branches?

Reply:
Added a note in this section. Let me know if that doesn't resolve it.